File: 9antrakidsFormationBack/src/main/java/com/esprit/springjwt/entity/Session.java
package com.esprit.springjwt.entity;


import java.io.Serializable;
import java.util.*;

import javax.persistence.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;
@Entity
@Data
public class Session implements Serializable {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long id;

    private String SessionName ;
    private String Description ;
	@Temporal(TemporalType.TIMESTAMP)
    private Date startDate;

	@Column(unique = true)
	private String GeneratedLink;
	@Temporal(TemporalType.TIMESTAMP)
    private Date finishDate;
    private String GroupSession;

    @ManyToMany
    @JoinTable(
        name = "session_group",
        joinColumns = @JoinColumn(name = "session_id"),
        inverseJoinColumns = @JoinColumn(name = "group_id")
    )
    @JsonIgnore
    private List<Groups> groups = new ArrayList<>();
    @ManyToOne
    @JoinColumn(name = "formation_id")
    private Formation formation;

	@ElementCollection
	@CollectionTable(name = "user_presence_statuss", joinColumns = @JoinColumn(name = "group_id"))
	@MapKeyColumn(name = "user_id")
	@Column(name = "presence_status")
	private Map<Long, Boolean> userPresence = new HashMap<>();



}

File: 9antraFormationFrant/src/app/Models/Session.ts
import { Formation } from './formation.model';
import { Groups } from './group.model';

export class Session {
  id?: number;
  sessionName?: string;
  description?: string;
  startDate!: Date;
  finishDate!: Date;
  groupSession?: string;
  groups: Groups[] = [];
  formation!: Formation;
  userPresence!: { [key: number]: boolean }; // Represents the map of userId to presence status

  GeneratedLink?: string;
}
function mapToObject(map: Map<number, boolean>): { [key: number]: boolean } {
  const obj: { [key: number]: boolean } = {};
  map.forEach((value, key) => {
    obj[key] = value;
  });
  return obj;}